requirements/datasets.txt

```diff
@@ -0,0 +1,18 @@
+# This file is managed by dependabot and should not be manually edited unless you need
+# to add or remove a dependency. It documents the version of our dependencies used in
+# CI, these are not hard requirements for using TorchGeo.
+
+# datasets
+h5py==3.6.0
+laspy==2.1.2
+open3d==0.14.1;python_version<'3.10'
+opencv-python==4.6.0.66
+pandas==1.4.2;python_version>='3.8'
+pandas==1.3.5;python_version=='3.7'
```

Comment:
It's still unclear if dependabot will actually update lines like this, or if it will try to update both. Ideally, it would only update the first occurrence, or somehow parse the environment marker. If that doesn't work, we can create separate files for Python 3.7 (and 3.8 once numpy and friends drop that too).